src/backend/api/meals.js

```diff
@@ -4,12 +4,76 @@ const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if ("maxPrice" in request.query) {
+      const maxPrice = request.query.maxPrice
+      const meals = await knex("meal").where("price", "<=", maxPrice)
+      response.send(meals)
+      return
+    }
+    if ("availableReservations" in request.query && request.query.availableReservations == "true") {
+      const meals = await knex.raw(
+        `SELECT meal.id, meal.title, meal.max_reservations, SUM(reservation.number_of_guests) AS reserved
+          FROM meal
+          LEFT JOIN reservation
+          ON meal.id = meal_id
+          GROUP BY meal_id
+          HAVING max_reservations > reserved OR reserved IS NULL;
+          `)
+      console.log(meals)
+      response.send(meals)
+      return
+    }
+    if ("title" in request.query) {
+      const title = request.query.title
+      const meals = await knex("meal").where("title", "like", "%" + title + "%")
+      response.send(meals)
+      return
+    }
+    if ("createdAfter" in request.query) {
+      const createdAfter = Data.parse(request.query.createdAfter)
+      const meals = await knex("meal").where("created_date", ">=", createdAfter)
+      response.send(meals)
+      return
+    }
+    if ("limit" in request.query) {
+      const limit = Number(request.query.limit)
+      const meals = await knex("meal").limit(limit)
+      response.send(meals)
+      return
+    }
+    const meals = await knex("meal")
```

Comment:
Good work in using knex to query the database and filter the meals directly instead of loading them and using the filter method.
One thing though that will not work, is multiple query parameters For instance, if I wanted to see meals with a certain title AND also created after a date, I will only get results based on my title parameter!